core/src/subject/SubjectEntity.ts

```diff
@@ -49,75 +231,192 @@ export class SubjectEntity {
     return this.#perspective;
   }
 
-  private async getData(id?: string) {
-    const tempId = id ?? this.#baseExpression;
-    let data = await this.#perspective.getSubjectData(this.#subjectClass, tempId)
-    Object.assign(this, data);
-    this.#baseExpression = tempId;
-    return this
+  private static async assignValuesToInstance(
+    perspective: PerspectiveProxy,
+    instance: SubjectEntity,
+    values: ValueTuple[]
+  ) {
+    // Map properties to object
+    const propsObject = Object.fromEntries(
+      await Promise.all(
+        values.map(async ([name, value, resolve]) => {
+          // Resolve the value if necessary
+          const finalValue = resolve ? (await perspective.getExpression(value)).data : value;
```

Comment:
So if we have the literal handling in Prolog as above, we will need to understand here that we don't have to resolve those values. Maybe we should have an internal `ResolveInternal` prolog variable above and only if it's not a literal language URI set `Resolve = true`...